### PR TITLE
docs(react-native): UI Cookbook - Connection Quality Indicator

### PR DESCRIPTION
**Pull request by khushal87**

### CODE REVIEW

**Review by MartinCupela — APPROVED** [review]

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx [hunk 1]
```diff
@@ -3,7 +3,88 @@ id: connection-quality-indicator
...
-:::
+Network conditions are not always perfect so instead of playing guessing games as to why is someone's video so blurry our default [`ParticipantView`](../../ui-components/participants/participant-view/) component comes equipped with connection quality indicator which provides general information about that participants network conditions in four states:
+
+- UNSPECIFIED
+- POOR
+- GOOD
+- EXCELLENT
+
+![Preview of the Connection quality indicator](../assets/06-advanced/10-connection-quality-indicator/connection-quality-indicator.png)
+
```
Comment (santhoshvai) [inline packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:14]: small nitpick on the patch, it should not be inside 06-advanced but it should be inside 05-ui-cookbook

very nice screenshot though

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx [hunk 2]
```diff
@@ -3,7 +3,88 @@ id: connection-quality-indicator
...
+}: {
+  participant: StreamVideoParticipant | StreamVideoLocalParticipant;
+}) => {
+  const readableConnectionQuality =
+    SfuModels.ConnectionQuality[participant.connectionQuality];
+
+  return (
+    <Text>
+      {`${readableConnectionQuality} -
+      ${'⭐️'.repeat(participant.connectionQuality)}`}
```
Comment (santhoshvai) [inline packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:55]: is the screenshot and the indicator matching.. here we show stars ?!

**Review by santhoshvai — CHANGES_REQUESTED** [review]
the screenshot and code have a mismatch it seems

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx (hunk 2 above)
Comment (khushal87) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:55]: The screen shot above has no relation with the code snippet. We show the general component above and here we show how you can achieve or make a similar component and mostly focuses on what data to use while building such component.

Comment (santhoshvai) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:55]: ``${readableConnectionQuality} -` this should be removed in my view 

`${'⭐️'.repeat(participant.connectionQuality)}` is sufficient in my view

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx [hunk 3]
```diff
@@ -3,7 +3,88 @@ id: connection-quality-indicator
...
-If you have any questions, please reach out to us on our [website](https://getstream.io/video/#contact).
-:::
+Network conditions are not always perfect so instead of playing guessing games as to why is someone's video so blurry our default [`ParticipantView`](../../ui-components/participants/participant-view/) component comes equipped with connection quality indicator which provides general information about that participants network conditions in four states:
+
+- UNSPECIFIED
+- POOR
+- GOOD
+- EXCELLENT
+
+![Preview of the Connection quality indicator](../assets/06-advanced/10-connection-quality-indicator/connection-quality-indicator.png)
```
Comment (santhoshvai) [inline packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:13]:
```suggestion
![Preview of the default Connection quality indicator](../assets/06-advanced/10-connection-quality-indicator/connection-quality-indicator.png)
```

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx (hunk 2 above)
Comment (santhoshvai) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:55]: I get it.. what I meant is can we add a screenshot for the one here or do you think its unnecessary

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx (hunk 3 above)
Comment (santhoshvai) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:13]: can we also add a caption on the image here please that's more clear

https://staging.getstream.io/video/docs/react/ui-cookbook/custom-label/

Comment (khushal87) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:13]: It is done using the ImageShowcase component and which brings the image to the left by default which doesn't look good on the page.

Comment (santhoshvai) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx]: can we use this image instead

https://staging.getstream.io/video/docs/assets/images/network-quality-indicator-a44a4b93a1253e9ca87f030774096385.png

Comment (khushal87) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx]: ImageShowcase is used when there are multiple images to show

Comment (santhoshvai) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx]: If not using ImageShowcase lets add a sentence above saying that this is a preview of the default component please

Comment (khushal87) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx]: Done

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx (hunk 2 above)
Comment (khushal87) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx]: Done

File: packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx [hunk 4]
```diff
@@ -3,7 +3,97 @@ id: connection-quality-indicator
...
+
+const readableConnectionQuality =
+  SfuModels.ConnectionQuality[participant.connectionQuality];
+
+console.log(readableConnectionQuality); // -> EXCELLENT
+```
+
+## Custom Connection quality indicator component
+
+You'll most likely be displaying this indicator component inside each participant view ([ParticipantView](../../ui-components/core/participant-view)) within a call layout, you can learn more about `ParticipantView` customizations in [the participant view customizations guide](../participant-view-customizations).
```
Comment (santhoshvai) [inline packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx:34]:
```
you can learn more about `ParticipantView` customizations in [the participant view customizations guide](../participant-view-customizations).
```

probably we wont have this now

Comment (khushal87) [reply packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/10-connection-quality-indicator.mdx]: Fixed

**Pull request merged**